src/hdmf/container.py

```diff
@@ -801,6 +813,19 @@ def set_dataio(self, **kwargs):
         dataio.data = self.__data
         self.__data = dataio
 
+    def set_data_io(self, data_io_class: Type[DataIO], data_io_kwargs: dict) -> None:
```

Comment:
Can you add a deprecation warning to `Data.set_dataio` and tell people to use `Data.set_data_io` instead?

Reply:
done